src/main/java/com/cloudbees/hudson/plugins/folder/buildable/DisableAction.java

```diff
@@ -0,0 +1,125 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.cloudbees.hudson.plugins.folder.buildable;
+
+import com.cloudbees.hudson.plugins.folder.Folder;
+import com.cloudbees.hudson.plugins.folder.Messages;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.Ordering;
+import hudson.Extension;
+import hudson.model.AbstractProject;
+import hudson.model.Action;
+import hudson.model.Job;
+import hudson.util.HttpResponses;
+import jenkins.model.TransientActionFactory;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.logging.Logger;
+
+public class DisableAction implements Action {
+    private final Folder item;
+
+    public DisableAction(Folder folder) {
+        this.item = folder;
+    }
+
+    public String getIconFileName() {
+        if (!item.hasPermission(Job.CONFIGURE) || item.isDisabled()) {
+            return null;
+        }
+        return "disabled.png";
+    }
+
+    public String getDisplayName() {
+        return Messages.DisableAction_displayName();
+    }
+
+    public String getUrlName() {
+        return "disable";
+    }
+
+    public Folder getItem() {
+        return item;
+    }
+
+    public Collection<? extends Job> getJobs() {
+        return new Ordering<Job>() {
+            @Override
+            public int compare(Job job, Job t1) {
+                if (job.getParent().equals(t1.getParent())) {
+                    return job.getName().compareTo(t1.getName());
+                }
+                return job.getParent().getFullDisplayName().compareTo(t1.getParent().getFullDisplayName());
+            }
+        }.immutableSortedCopy(getPossibleJobs());
+    }
+
+    private Collection<? extends Job> getPossibleJobs() {
+        return Collections2.filter(item.getAllJobs(), new Predicate<Job>() {
+            public boolean apply(Job job) {
+                return (job.isBuildable() && job.hasPermission(Job.CONFIGURE));
+            }
+        });
+    }
+
+    /**
+     * Disable jobs contained in this folder.
+     */
+    @RequirePOST
+    public HttpResponse doDisable(StaplerRequest req) throws IOException {
+        LOGGER.finest("Trying to disable all jobs in " + item.getName());
+        for (Job job : getPossibleJobs()) {
+            try {
+                ((AbstractProject) job).disable();
+                LOGGER.finest("Set " + job.getName() + " as not buildable");
+            } catch (ClassCastException e) {
+                LOGGER.info("Cannot disable " + job.getName());
```

Comment:
CCE is not something a user cares about, it just means the API for doing this is weird. This should be handled internally, and displayed to the user as "The following cannot be disabled".